resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -91,6 +91,9 @@ private[yarn] class YarnAllocator(
   @GuardedBy("this")
   private val releasedContainers = collection.mutable.HashSet[ContainerId]()
 
+  @GuardedBy("this")
+  private val completedContainerIds = collection.mutable.HashSet[ContainerId]()
```

Comment:
you never remove anything from this?  Seems like we are adding a bunch of extra memory pressure for very long running applications using dynamic allocation that might have a lot of containers.